### PR TITLE
Run forbidden api checks with runtimeJavaVersion

### PR DESCRIPTION
**Pull request by alpar-t**
Switching to the CLI version of the tool allows us to run it with the runtime java version. 
The newly introduced task just a  thin wrapper over the cli args and we configure it as before from precommit. Since we only used it in one place, the extensions is no longer used as `tasks.withType` achieves the same goal.

### CODE REVIEW

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/PrecommitTasks.groovy
Comment (atorok) [inline buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/PrecommitTasks.groovy]: Will wait for the PR fixing the violations to be merged before changing this and merging this PR.

File: buildSrc/src/main/java/org/elasticsearch/gradle/precommit/ForbiddenApisCliTask.java
Comment (atorok) [inline buildSrc/src/main/java/org/elasticsearch/gradle/precommit/ForbiddenApisCliTask.java]: This avoids inheriting from `JavaExec`. I prefer that because this way the task can control when to do the exec and we can do the required setup in the task action.
Gradle runs the task action of the inherited task first ( I couldn't find a documented order ).

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/PrecommitTasks.groovy
Comment (rjernst) [reply buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/PrecommitTasks.groovy]: I think this should always be compiler version? That way if eg we use java 11, we will detect problems ahead of time, before we switch to using java 11 all the time?

File: buildSrc/src/main/java/org/elasticsearch/gradle/precommit/ForbiddenApisCliTask.java
Comment (rjernst) [inline buildSrc/src/main/java/org/elasticsearch/gradle/precommit/ForbiddenApisCliTask.java]: While I know we still need forbidden on the gradle classpath because we invoke it through ant for third-party-audit, I think once we move that to use cli, we should change this to have forbidden api in a separate configuration and pull it there, instead of this hack on finding where a class we have already loaded is located.

File: buildSrc/src/main/java/org/elasticsearch/gradle/ClassPathUtils.java
Comment (rjernst) [inline buildSrc/src/main/java/org/elasticsearch/gradle/ClassPathUtils.java]: I don't like this hack, and especially not making it easier to use. Can it be isolated, or at least given large warning comments to not use this?

File: buildSrc/src/main/java/org/elasticsearch/gradle/ClassPathUtils.java
Comment (rjernst) [inline buildSrc/src/main/java/org/elasticsearch/gradle/ClassPathUtils.java]: typo: bath -> path

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/PrecommitTasks.groovy
Comment (rjernst) [inline buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/PrecommitTasks.groovy]: Can we stick with the dsl of forbidden apis, instead of adding these helpers? I would like us to be able to switch back if/when it supports forking.

File: buildSrc/src/main/java/org/elasticsearch/gradle/precommit/NamingConventionsTask.java
Comment (rjernst) [inline buildSrc/src/main/java/org/elasticsearch/gradle/precommit/NamingConventionsTask.java]: I would prefer if we kept this url hack as it was and didn't make it easier to use

**Review by rjernst — COMMENTED** [review]
Thanks @atorok. I have a few comments.

**Comment by atorok** [comment]
@rjernst I reverted the hack, and wired it to compiler version.

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/PrecommitTasks.groovy
Comment (rjernst) [inline buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/PrecommitTasks.groovy]: Should the forbidden apis configuration also be an input?

File: buildSrc/src/main/java/org/elasticsearch/gradle/precommit/ForbiddenApisCliTask.java
Comment (rjernst) [inline buildSrc/src/main/java/org/elasticsearch/gradle/precommit/ForbiddenApisCliTask.java]: nit: this method name does not seem accurate, since it does the actual execution, not just writing the marker file

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/PrecommitTasks.groovy
Comment (rjernst) [reply buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/PrecommitTasks.groovy]: Alright, that is fine. Can we please name it `set` instead of `replace` then, to match common naming with other gradle dsl elements?

**Review by rjernst — APPROVED** [review]
LGTM. A couple last nits.

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/PrecommitTasks.groovy
Comment (atorok) [reply buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/PrecommitTasks.groovy]: `set` will not work, as the task already has that method, so the extension will never be fetched and called, but even with something like `setSignatures` it would be confusing as I don't think `signatures = ...` would work, as it won't call the closure on task extensions but look for a set method on the task instance.

Comment (atorok) [reply buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/PrecommitTasks.groovy]: I'm going to merge and submit a PR for the third party audit as well.
Let me know and I'll make changes to the name if needed,

**Pull request merged**
